### ISSUE TITLE
Solving too many problems at once

### ISSUE THREAD

**Issue by edygar** [leonardorb/react-tree-component#1]
Constructive critic here:
You are solving too many problems at once here. By fetching data instead of dealing just with the presentation you are locking both the tree and the data source.

IMHO, a better approach could be decoupling the data fetching from the tree state component. It could deal only with the state of the tree, what is opened or not, select/focused or not, otherwise it's a too smart dumb component. Where the fetched data lives? Right now, it is seized into component's state.

**Comment by leonardorb** [comment]
@edygar you're totally right. This is included in our bullet points in the `README` (under `On the road map`).
- A more generic approach

Data fetching does not belong to the component at all. In a near future, everything that the component will need to know is where is the data and what tree node param we're using to control and build hierarchy.

Thanks for your comment! And I'm expecting some contributions along the way. :)

**Comment by edygar** [comment]
```
Thanks for your comment! And I'm expecting some contributions along the way. :)
```

Sure, I'm very enthusiastic about React, and it's always nice to put it in practice.

Have you seen https://github.com/chenglou/react-treeview ? To be honest, I didn't like its API but using a recursive Component is pretty appealing. Another interesting point to note is that `Node Collapse` state  might reside both in ou outside of the container, as in uncontrolled and controlled examples.
